### DISCUSSION TITLE
PICO W Bluetooth not working

### DISCUSSION THREAD

**Discussion by JH2k00** [micropython/micropython#12616] [RP2040 / Pico]
Greetings,

I'm currently working on a project that requires bluetooth and ulab (numpy/scipy for micropython) on the Raspberry Pi Pico W. To this effect, I downloaded the most recent version of micropython with ulab integrated from https://github.com/v923z/micropython-builder/releases, which has been updated 8 hours ago, so it should contain the latest micropython version. With this firmware however, the following code snippet never gets to printing "Done" : 
```py
from micropython import const

import aioble
import bluetooth

# org.bluetooth.service.pulse_oximeter
_PLX_SENSE_UUID = bluetooth.UUID(0x1822)

# org.bluetooth.characteristic.heart_rate_measurement
_PLX_CHAR_HR_UUID = bluetooth.UUID(0x2A37)

# org.bluetooth.characteristic.gap.appearance.xml
_ADV_APPEARANCE_GENERIC_PULSE_OXIMETER = const(0x0C40)

# How frequently to send advertising beacons.
_ADV_INTERVAL_MS = 250000

# Register GATT server.
plx_service = aioble.Service(_PLX_SENSE_UUID)
plx_characteristic = aioble.Characteristic(
    plx_service, _PLX_CHAR_HR_UUID, read=True, notify=True
)
print("Registering")
aioble.register_services(plx_service)
print("Done")
```

While trying to resolve the issue, I downloaded the latest micropython release for the Raspberry Pi Pico W from https://micropython.org/download/RPI_PICO_W/ , and ran the same code with the new firmware. The code executed flawlessly and got to printing "Done". 

[all_firmware.zip](https://github.com/micropython/micropython/files/12837745/all_firmware.zip)

I then tried to compile my own firmware for the RPI PICO W using the instruction at https://github.com/micropython/micropython/tree/master/ports/rp2 and the v1.21.0 tag , in order to check if the problem persists, or if it's exclusive to ulab. The code unfortunately never got to printing "Done" with the new self-compiled firmware (which doesn't contain ulab), leading me to think that the instructions for compiling the firmware may be uncomplete, as the official released version works fine in contrast to the self-compiled one. 

I've zipped and attached all 3 versions of the firmware to this message. I hope that I've provided enough information, and thank you all for your time and help in advance. 

Best regards,
JH

**Comment by andrewleech** [comment]
Duplicate of https://github.com/micropython/micropython/issues/12617
